src/reanimated2/interpolation.js

```diff
@@ -1,30 +1,115 @@
 import { Extrapolate } from '../derived/interpolate';
 
+function getVal(config) {
+  'worklet';
+
+  const { type, coef, val, ll, rr, x } = config;
+
+  switch (type) {
+    case Extrapolate.IDENTITY:
+      return x;
+    case Extrapolate.CLAMP:
+      if (coef * val < coef * ll) {
+        return ll;
+      }
+      return rr;
+    case Extrapolate.EXTEND:
+    default:
+      return val;
+  }
+}
+
+function isExtrapolate(value) {
+  'worklet';
+
+  return (
+    value !== Extrapolate.EXTEND &&
+    value !== Extrapolate.CLAMP &&
+    value !== Extrapolate.IDENTITY
+  );
+}
+
+function validateType(type) {
+  'worklet';
+
+  if (
+    (typeof type === 'object' &&
+      (!type.extrapolateLeft &&
+        !type.extrapolateRight &&
+        Object.keys(type).length >= 1 &&
+        type.constructor === Object)) ||
+    (Object.keys(type).length > 2 && type.constructor === Object)
+  ) {
+    throw new Error(
+      `Reanimated: config object is not valid please provide valid config, for example:
+       interpolate(value, [inputRange], [outputRange], {
+        extrapolateLeft: 'clamp',
+        extrapolateRight: 'extend',
+      })`
+    );
+  }
+
+  if (
+    typeof type === 'object' &&
+    type.extrapolateLeft &&
+    isExtrapolate(type.extrapolateLeft)
+  ) {
+    throw new Error(
+      `Reanimated: not supported value for "extrapolateLeft" \nSupported values: ["extend", "clamp", "identity"]\n Valid example:
+       interpolate(value, [inputRange], [outputRange], {
+        extrapolateLeft: 'clamp',
+      })`
+    );
+  }
+
+  if (
+    typeof type === 'object' &&
+    type.extrapolateRight &&
+    isExtrapolate(type.extrapolateRight)
+  ) {
+    throw new Error(
+      `Reanimated: not supported value for "extrapolateRight" \nSupported values: ["extend", "clamp", "identity"]\n Valid example:
+       interpolate(value, [inputRange], [outputRange], {
+        extrapolateRight: 'clamp',
+      })`
+    );
+  }
+
+  if (typeof type === 'string' && isExtrapolate(type)) {
+    throw new Error(
+      `Reanimated: not supported value for "interpolate" \nSupported values: ["extend", "clamp", "identity"]\n Valid example:
+       interpolate(value, [inputRange], [outputRange], "clamp")`
+    );
+  }
+}
+
 function internalInterpolate(x, l, r, ll, rr, type) {
   'worklet';
   if (r - l === 0) return ll;
   const progress = (x - l) / (r - l);
   const val = ll + progress * (rr - ll);
   const coef = rr >= ll ? 1 : -1;
 
+  const config = { type, coef, val, ll, rr, x };
+
+  validateType(type);
+
   // TODO: support default values in worklets:
   // e.g. function interplate(x, input, output, type = Extrapolate.CLAMP)
-  type = type || Extrapolate.EXTEND;
+  // type = type || Extrapolate.EXTEND;
```

Comment:
Why it's commented out? If we need default then it should be left and if we don't please just remove this value.

BTW, does it break backward compat?

Reply:
> Why it's commented out? If we need default then it should be left and if we don't please just remove this value.

Removed

---

> BTW, does it break backward compat?

If you mean using it like this -> ```interpolate(translation.y.value, [0, -75], [0, -150], 'clamp')``` 

It will work like before 🙂